src/imputation/MoR.py

```diff
@@ -0,0 +1,85 @@
+"""Functions for the Mean of Ratios (MoR) methods."""
+import pandas as pd
+import re
+
+from src.imputation.tmi_imputation import apply_to_original
+from src.imputation.apportionment import run_apportionment
+
+good_statuses = ["Clear", "Clear - overridden"]
+bad_statuses = ["Form sent out", "Check needed"]
+
+
+def run_mor(df, backdata, target_vars):
+    """Function to implement Mean of Ratios method.
+
+    This is implemented by first carrying forward data from last year
+    for non-responders, and then calculating and applying growth rates
+    for each imputation class.
+
+    Args:
+        df (pd.DataFrame): Processed full responses DataFrame
+        backdata (pd.DataFrame): One period of backdata.
+        target_vars ([string]): List of variables to impute.
+
+    Returns:
+        pd.DataFrame: df with MoR applied
+    """
+    to_impute_df, backdata = mor_preprocessing(df, backdata)
+
+    # Carry forwards method
+    carried_forwards_df = carry_forwards(to_impute_df, backdata, target_vars)
+    carried_forwards_df["imp_marker"] = "CF"
+
+    # TODO Remove the `XXX_prev` columns (left in for QA)
+    return apply_to_original(carried_forwards_df, df)
+
+
+def mor_preprocessing(df, backdata):
+    """Apply pre-processing ready for MoR
+
+    Args:
+        df (pd.DataFrame): full responses for the current year
+        backdata (pd.Dataframe): backdata file read in during staging.
+    """
+    # Select only values to be imputed and remove duplicate instances
+    to_impute_df = df.copy().loc[
+        (
+            (df["formtype"] == "0001")
+            & (df["status"].isin(bad_statuses))
+            & (df["instance"] == 0 | pd.isnull(df["instance"]))
```

Comment:
The only way we'd get a null instance is for "form sent out" and in TMI we create these as instance 1 (long forms only)
I suggest we add in a TODO here to move this to imputation_main.